src/core/positioning/ntripsocketclient.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <QObject>
+#include <QTcpSocket>
+#include <QTimer>
+
+class NtripSocketClient : public QObject
+{
+    Q_OBJECT
+  public:
+    explicit NtripSocketClient( QObject *parent = nullptr );
+    ~NtripSocketClient();
+
+    qint64 start(
+      const QString &host,
+      quint16 port,
+      const QString &mountpoint,
+      const QString &username,
+      const QString &password );
+
+    void stop();
+
+  signals:
+    void correctionDataReceived( const QByteArray &data );
+    void errorOccurred( const QString &message );
+    void streamConnected();
```

Comment:
We should have a streamDisconnected signal to allow for the code to react to a server-induced disconnection.